src/engine/Interfaces/LifecycleEvents.ts

```diff
@@ -88,11 +88,18 @@ export interface CanInitialize {
   off(eventName: Events.initialize, handler?: (event: Events.InitializeEvent<any>) => void): void;
 }
 
-export interface CanActivate {
+export interface SceneActivationContext<TData = undefined> {
+  data?: TData;
+  previousScene: Scene;
```

Comment:
we could alternatively call this `prevScene` for convenience

Reply:
`previousScene` is good, we try to avoid too many abreviations 👍